src/main/java/com/unstoppabledomains/util/DnsUtils.java

```diff
@@ -0,0 +1,117 @@
+import com.google.gson.JsonSyntaxException;
+import com.unstoppabledomains.exceptions.DnsException;
+import com.unstoppabledomains.exceptions.DnsExceptionCode;
+import com.unstoppabledomains.exceptions.NSExceptionParams;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class DnsUtils {
+  
+  static Number DEFAULT_TTL = 300;
+  public List<DnsRecord> toList(Map<String, String> rawRecords) throws DnsException {
+    List<DnsRecord> dnsRecords = new ArrayList<DnsRecord>();
+    List<DnsRecordsType> types = getAllDnsTypes(rawRecords);
+    for (DnsRecordsType type: types) {
+      List<DnsRecord> dnsType = constructDnsRecords(rawRecords, type);
+      dnsRecords.addAll(dnsType);
+    }
+    return dnsRecords;
+  }
+
+  public Map<String, String> toMap(List<DnsRecord> records) throws DnsException {
+    Map<String, String> map = new HashMap<String, String>();
+    for (DnsRecord record: records) {
+      DnsRecordsType type = record.getType();
+      String ttlInRecord = map.get("dns." + type + ".ttl");
+      JsonArray dnsInRecord = getJsonArray(map.get("dns." + type), type);
+      if (dnsInRecord != null) {
+        dnsInRecord.add(record.getData());
+        map.put("dns." + type, dnsInRecord.toString());
+      } else {
+        map.put("dns." + type, "["+record.getData()+"]");
+        map.put("dns." + type + ".ttl", record.getTtl().toString());
+      }
+
+      if (ttlInRecord != null && !ttlInRecord.isEmpty() && !ttlInRecord.equals(record.getTtl().toString())) {
+        throw new DnsException(DnsExceptionCode.InconsistentTtl, new NSExceptionParams("r", type.toString()));
+      }
+    }
+    return map;
+  }
+
+  private JsonArray getJsonArray(String jsonArray, DnsRecordsType type) throws DnsException {
+    try {
+      return JsonParser.parseString(jsonArray).getAsJsonArray();
+    } catch(NullPointerException exception) {
+      // this is possible in toMap function only
+      return null;
+    } catch(JsonSyntaxException exception) {
+      throw new DnsException(DnsExceptionCode.DnsRecordCorrupted,
+        new NSExceptionParams("r", type.toString()));
+    }
+  }
+
+  private List<DnsRecordsType> getAllDnsTypes(Map<String, String> rawRecords) {
+    Set<DnsRecordsType> dnsTypes = new HashSet<DnsRecordsType>();
+    for (Map.Entry<String, String> entry : rawRecords.entrySet()) {
+      String[] chunks = entry.getKey().split("\\.");
+      Boolean isDnsType = chunks[0].equals("dns") && !chunks[1].equals("ttl");
+      if (isDnsType.equals(Boolean.TRUE)) {
+        dnsTypes.add(DnsRecordsType.valueOf(chunks[1]));
+      }
+    }
+    return new ArrayList<DnsRecordsType>(dnsTypes);
+  }
+
+  private List<DnsRecord> constructDnsRecords(Map<String, String> rawRecords, DnsRecordsType type) throws DnsException {
+    Number ttl = parseTtl(rawRecords, type);
+    String jsonValueString = rawRecords.get("dns." + type.toString());
+    if (jsonValueString.isEmpty()) {
```

Comment:
```suggestion
    if (StringUtils.isEmpty(jsonValueString)) {
```
This will guard you against potential `NullPointerException`.